dev/src/cli/cli_create.ts

```diff
@@ -0,0 +1,262 @@
+    lines.push(`GOOGLE_GENAI_USE_VERTEXAI=0`);
+  }
+  if (options.project) {
+    lines.push(`GOOGLE_CLOUD_PROJECT=${options.project}`);
+  }
+  if (options.region) {
+    lines.push(`GOOGLE_CLOUD_LOCATION=${options.region}`);
+  }
+  if (options.region && options.project) {
+    lines.push(`GOOGLE_GENAI_USE_VERTEXAI=1`);
+  }
+  await saveToFile(dotEnvPath, lines.join('\n'));
+}
+
+export async function createAgent(options: AgentCreationOptions) {
+  const agentDir = path.join(dirname, options.agentName);
+  await generateAgentFolder(agentDir);
+
+  if (!options.model) {
+    const model: symbol | string = (await select(
+      {
+        message: 'Choose a model for the root agent',
+        options: [
+          { label: 'gemini-2.5-flash', value: 'gemini-2.5-flash' },
+          { label: 'gemini-2.5-pro', value: 'gemini-2.5-pro' },
+          { label: 'gemini-3-pro-preview', value: 'gemini-3-pro-preview' },
+          { label: 'gemini-3-flash-preview', value: 'gemini-3-flash-preview' },
+        ],
+      },
+    ));
+    if(isCancel(model)) {
+      process.exit(0);
+    }
+    options.model = model;
+  }
+  
+
+  if(options.language !== "js" && options.language !== "ts") {
+    const language = (await select({
+        message: 'Choose a language for the agent',
+        options: [
+          { label: 'JavaScript', value: 'js' },
+          { label: 'TypeScript', value: 'ts' },
+        ],
+      },
+    ));
+    if(isCancel(language)) {
+      process.exit(0);
+    }
+    options.language = language;
+  }
+
+  if (!options.apiKey && !options.project) {
+    const backend: symbol | string = (await select({
+        message: 'Choose a backend',
+        options: [
+          { label: 'Google AI', value: 'googleai' },
+          { label: 'Vertex AI', value: 'vertex' },
+        ],
+      },
+    ));
+    if(isCancel(backend)) {
+      process.exit(0);
+    }
+
+    if (backend === 'vertex') {
+      const projectResponse: string = (await text(
+        {
+          message: 'Enter the Google Cloud Project ID',
+          initialValue: await getGcpProject(),
+        },
+      )).toString();
+      if(isCancel(projectResponse)) {
+        process.exit(0);
+      }
+      options.project = projectResponse;
+
+      const regionResponse: symbol | string = (await text({
+          message: 'Enter the Google Cloud Region',
+          initialValue: await getGcpRegion(),
+        },
+      ));
+      if(isCancel(regionResponse)) {
+        process.exit(0);
+      }
+      options.region = regionResponse;
+    } else {
+      const apiKeyResponse: symbol | string = (await text(
+        {
+          message: 'Enter the Google API Key',
+        },
+      ));
+      if(isCancel(apiKeyResponse)) {
+        process.exit(0);
+      }
+      options.apiKey = apiKeyResponse;
+    }
+  }
+
+  await generateFiles(options);
+  await saveToFile(path.join(agentDir, 'package.json'), PACKAGE_JSON(options.agentName, options.language));
```

Comment:
now this is a file generation command also. can you please move it to the generateFiles function?